### PR TITLE
Added hover effects in Footer icons in the website

### LINKED ISSUE #716: [feat] The footer socials icon hover experience can be enhanced
### Would you like to have a new feature? Please describe.

The footer of the [WeMakeDevs website](https://wemakedevs.org/) has many social icons, but, when I hover the mouse pointer, the icons still remain as it is without any effects in them. So, I think the hover experience can be made better.
### Effects that can be added to the icons :
- On hovering, the icons can have colour specific to the social branding colour
- On hovering, the icons can have **ease-in-out** _transition_ with a **delay of around 200-300 seconds**

### How would this feature be beneficial for the project

It might have a good and cool impression on the end user. So, it effectively makes the Website footer visually more appealing.

### Additional context

I have provided below a video describing the effects I have mentioned.

https://github.com/WeMakeDevs/wemakedevs/assets/105960032/af32b05f-a705-439d-b93e-4e4661ddefca

### PR DESCRIPTION
**Pull request by SaptarshiSarkar12**
## Fixes Issue

This PR fixes the following issues:

- closes #716 

## Changes proposed

Here comes all the changes proposed through this PR
- Added colours of the logo of socials to the icons in the WeMakeDevs website's footer.

## Check List (Check all the boxes which are applicable) 

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

Add all the screenshots which support your changes

https://github.com/WeMakeDevs/wemakedevs/assets/105960032/25f645a4-97aa-4450-a75f-a55daaf53112

### CODE REVIEW

**Comment by SaptarshiSarkar12** [comment]
I need some help on how to add hover effects @Chandraprakash-Darji @AnuragThePathak.

**Comment by Shiva953** [comment]
@SaptarshiSarkar12 for hover effect
try adding tailwind classes
```
className=`text-2xl transition-all duration-200 ease-in-out delay-200 hover:-translate-y-1`
```
[here](https://github.com/WeMakeDevs/wemakedevs/blob/e3985947b5d37a21f3671ecc49cf6fafb1a2b176/src/components/layout/Footer.jsx#L78C11-L89C17)

**Comment by SaptarshiSarkar12** [comment]
> @SaptarshiSarkar12 for hover effect try adding tailwind classes
> 
> ```
> className=`text-2xl transition-all duration-200 ease-in-out delay-200 hover:-translate-y-1`
> ```
> 
> [here](https://github.com/WeMakeDevs/wemakedevs/blob/e3985947b5d37a21f3671ecc49cf6fafb1a2b176/src/components/layout/Footer.jsx#L78C11-L89C17)

Yeah, hover effect works but colour disappears, don't know why. Could you please help @Shiva953 ?

**Comment by Shiva953** [comment]
> > @SaptarshiSarkar12 for hover effect try adding tailwind classes
> > ```
> > className=`text-2xl transition-all duration-200 ease-in-out delay-200 hover:-translate-y-1`
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > [here](https://github.com/WeMakeDevs/wemakedevs/blob/e3985947b5d37a21f3671ecc49cf6fafb1a2b176/src/components/layout/Footer.jsx#L78C11-L89C17)
> 
> Yeah, hover effect works but colour disappears, don't know why. Could you please help @Shiva953 ?

try replacing `{links.icon}` with `<span style={{ color: link.color }}>{link.icon}</span>`
and add a respective color property to each object in the `links` array
then edit `tailwind.config.js` and include the respective colours[this is the tricky bit though, colour themes are defined a bit differently here]

**Comment by SaptarshiSarkar12** [comment]
> > > @SaptarshiSarkar12 for hover effect try adding tailwind classes
> > > ```
> > > className=`text-2xl transition-all duration-200 ease-in-out delay-200 hover:-translate-y-1`
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > [here](https://github.com/WeMakeDevs/wemakedevs/blob/e3985947b5d37a21f3671ecc49cf6fafb1a2b176/src/components/layout/Footer.jsx#L78C11-L89C17)
> > 
> > 
> > Yeah, hover effect works but colour disappears, don't know why. Could you please help @Shiva953 ?
> 
> try replacing `{links.icon}` with `<span style={{ color: link.color }}>{link.icon}</span>` and add a respective color property to each object in the `links` array then edit `tailwind.config.js` and include the respective colours[this is the tricky bit though, colour themes are defined a bit differently here]

Hey @Shiva953, thank you for your ideas. I have tried tailwind css **text-[color]** property and it did work! But, I need a colour for the envelope icon, can you please suggest one?

**Comment by SaptarshiSarkar12** [comment]
@Shiva953 Here's the recording of the hover effect -

https://github.com/WeMakeDevs/wemakedevs/assets/105960032/a24e701b-6988-4a12-92c1-b32596e75e24

**Comment by Shiva953** [comment]
> > > > @SaptarshiSarkar12 for hover effect try adding tailwind classes
> > > > ```
> > > > className=`text-2xl transition-all duration-200 ease-in-out delay-200 hover:-translate-y-1`
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > [here](https://github.com/WeMakeDevs/wemakedevs/blob/e3985947b5d37a21f3671ecc49cf6fafb1a2b176/src/components/layout/Footer.jsx#L78C11-L89C17)
> > > 
> > > 
> > > Yeah, hover effect works but colour disappears, don't know why. Could you please help @Shiva953 ?
> > 
> > 
> > try replacing `{links.icon}` with `<span style={{ color: link.color }}>{link.icon}</span>` and add a respective color property to each object in the `links` array then edit `tailwind.config.js` and include the respective colours[this is the tricky bit though, colour themes are defined a bit differently here]
> 
> Hey @Shiva953, thank you for your ideas. I have tried tailwind css **text-[color]** property and it did work! But, I need a colour for the envelope icon, can you please suggest one?

try light yellow

**Comment by Shiva953** [comment]
> @Shiva953 Here's the recording of the hover effect -
> 
>  Screencast.from.2023-07-23.18-56-28.mp4

LGTM

**Comment by SaptarshiSarkar12** [comment]
Ok @Shiva953

**Comment by SaptarshiSarkar12** [comment]
@Shiva953 New Recording with the envelope color


https://github.com/WeMakeDevs/wemakedevs/assets/105960032/25f645a4-97aa-4450-a75f-a55daaf53112

**Comment by SaptarshiSarkar12** [comment]
@Shiva953 @Chandraprakash-Darji @AnuragThePathak Please review this PR.

**Comment by Shiva953** [comment]
> @Shiva953 @Chandraprakash-Darji @AnuragThePathak Please review this PR.

I think adding classnames as a property to the objects inside the links array makes it less reusable and chaotic. there must be a clean way of doing that with less and more re-usable code.

I was thinking of something similar to this : 
```
className={`text-2xl transition-all duration-200 ease-in-out delay-200 hover:-translate-y-1 hover:text-[${link.color}]`}
```

but it doesn't work
@Chandraprakash-Darji @AnuragThePathak thoughts?

**Comment by Chandraprakash-Darji** [comment]
@Shiva953 Tailwind compile on runtime, so we can't use template literal string are not valid as a className 

> `text-[${color}]` will not work

> you should use `style={{color}}`

File: src/components/layout/Footer.jsx [hunk 1]
```diff
@@ -162,53 +162,71 @@ const links = [
     link: 'https://www.youtube.com/@KunalKushwaha',
     icon: <FaYoutube />,
     ariaLabel: 'Visit WeMakeDevs on YouTube',
+    className:
+      'text-2xl transition-all delay-200 duration-200 ease-in-out hover:-translate-y-1 hover:text-[#FF0000]',
```
Comment (Chandraprakash-Darji) [inline src/components/layout/Footer.jsx:166]: ClassName should be like this
`className={clsxm( 'text-2xl transition-all delay-200 duration-200 ease-in-out hover:-translate-y-1', link.className)}`

And
In Object just ass the hover className

File: src/components/layout/Footer.jsx [hunk 2]
```diff
@@ -162,53 +162,71 @@ const links = [
...
+    className:
+      'text-2xl transition-all delay-200 duration-200 ease-in-out hover:-translate-y-1 hover:text-[#0088cc]',
   },
   {
     name: 'discord',
     link: 'https://discord.gg/wemakedevs',
     icon: <FaDiscord />,
     ariaLabel: 'Visit WeMakeDevs on Discord',
+    className:
+      'text-2xl transition-all delay-200 duration-200 ease-in-out hover:-translate-y-1 hover:text-[#46237a]',
```
Comment (Chandraprakash-Darji) [inline src/components/layout/Footer.jsx:222]: Use this for discord color `#7289DA`

**Review by Chandraprakash-Darji — CHANGES_REQUESTED** [review]

File: src/components/layout/Footer.jsx (hunk 1 above)
Comment (SaptarshiSarkar12) [reply src/components/layout/Footer.jsx]: Done @Chandraprakash-Darji ! Please re-review.

File: src/components/layout/Footer.jsx (hunk 2 above)
Comment (SaptarshiSarkar12) [reply src/components/layout/Footer.jsx]: Done @Chandraprakash-Darji !

**Comment by SaptarshiSarkar12** [comment]
@Chandraprakash-Darji Please review this PR.

**Comment by SaptarshiSarkar12** [comment]
@Chandraprakash-Darji @AnuragThePathak Please review this PR.

**Review by Chandraprakash-Darji — APPROVED** [review]

**Comment by SaptarshiSarkar12** [comment]
Thank you @Chandraprakash-Darji and @AnuragThePathak for the merge :grinning: !

**Pull request merged**